fundingmanager.go

```diff
@@ -1550,6 +1553,22 @@ func (f *fundingManager) handleFundingSigned(fmsg *fundingSignedMsg) {
 	// delete it from our set of active reservations.
 	f.deleteReservationCtx(peerKey, pendingChanID)
 
+	// Broadcast the finalized funding transaction to the network.
+	fundingTx := completeChan.FundingTxn
+	fndgLog.Infof("Broadcasting funding tx for ChannelPoint(%v): %v",
+		completeChan.FundingOutpoint, spew.Sdump(fundingTx))
+
+	err = f.cfg.PublishTransaction(fundingTx)
+	if err != nil {
+		fndgLog.Errorf("unable to broadcast funding "+
+			"txn: %v", err)
+		// We failed to broadcast the funding transaction, but watch
```

Comment:
If we fail to broadcast the transaction, depending on what the failure is, we should remove it from the database immediately. The internal implementation of `PublishTransaction` for btcwallet will remove the transaction from the txstore if it fails to broadcast.

Reply:
This is part of the issue we have atm, where if we can't ever get the node to accept our transaction, we account for the pending balance of the channel in `channelbalance` incorrectly as the utxo has already been marked as unspent again in the wallet internals.

Reply:
Would it be possible to still store the tx in the db, but in a failed open state? That way we can not count them towards the max pending channels, but still resume if somehow they do confirm (and prune if something else uses the utxo). Seems dangerous to delete them entirely, as we wouldn't have the information to recover

Reply:
I'm worried that just deleting the channel from the DB completely at this point could mean loss of funds if the transaction actually do confirm. If we did, we basically would trust the chain backend to guarantee that a failure here means that the tx cannot ever "leak" to the network.

A better approach is, as we talked about, at this point increase the fee of the funding tx and rebroadcast. We can then go on, with all the state needed intact.

Reply:
Yeah _ideally_ (assuming these other bugs are fixed), the only reason it's rejected is due to insufficient fees. With the fixes in btcwallet to allow replacement, etc we should no longer see cases where we accidentally double spend an input, or something similar.